src/combinatorics/young_diagram.lean

```diff
@@ -0,0 +1,93 @@
+/-
+Copyright (c) 2022 Jake Levinson. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Jake Levinson
+-/
+import order.upper_lower
+
+/-!
+# Young diagrams
+
+A Young diagram is a finite set of up-left justified boxes:
+
+[5, 3, 3, 1]
+□□□□□
+□□□
+□□□
+□
+
+Equivalently, a lower set in `ℕ × ℕ` in the product partial order. We write `(i, j) ∈ μ`
+to say that `(i, j)` (in matrix coordinates) is in the Young diagram `μ`.
+
+## Main functions:
+
+  - `young_diagram` : basic definition
+  - `young_diagram.card` : the cardinality (number of boxes)
+
+  - various constructors:
+      `young_diagram.has_empty` : the empty Young diagram
```

Comment:
```suggestion
      `young_diagram.has_emptyc` : the empty Young diagram
```
Not convinced this needs documenting.

Reply:
"`(∅ : young_diagram)` is the empty Young diagram" is worth documenting though

Reply:
Happy to take it out. (I was just trying to follow the instructions here and wasn't sure how much to include: https://leanprover-community.github.io/contribute/doc.html).